File: src/main/java/com/post/Box.java
package com.post;

public class Box {
    String name;
    float length;
    float width;
    float height;

    public Box(String name, float length, float width, float height) {
        this.name = name;
        this.length = length;
        this.width = width;
        this.height = height;
    }

    public boolean validate(float length, float width, float height) {
        if (length <= this.length && width <= this.width && height <= this.height) {
            return true;
        } else {
            return false;
        }
    }
}

File: src/main/kotlin/com/post/Box.kt
package com.kotlin

open class Box(val name: String, val length: Float, val width: Float, val height: Float) {
    open fun validate(length: Float, width: Float, height: Float) =
        (length <= this.length && width <= this.width && height <= this.height)
}

class Box3() : Box("Box3", 23f, 14f, 13f) {

}

class Box5() : Box("Box5", 39.5f, 27.5f, 23f) {

}